advocacy_docs/supported-open-source/patroni/migrating.mdx

```diff
@@ -0,0 +1,323 @@
+---
+title: 'Migrating to Patroni'
+description: "Migrating an existing Postgres cluster to Patroni"
+---
+
+This page describes how to migrate an existing Postgres cluster to Patroni managed cluster.
+
+### Overview of steps
+
+1. Disable current HA solution, if any.
+2. Perform the following steps on all nodes, starting with the primary and then going with standbys:
+    1. Disable Postgres systemd unit.
+    2. Create a YAML configuration file for Patroni. If you use replication slots in the existing cluster, then make sure to configure [permanent slots](https://patroni.readthedocs.io/en/latest/dynamic_configuration.html#:~:text=a%20standby%20leader-,slots,-%3A%20define%20permanent%20replication) so Patroni does not delete them while migrating.
+    3. Enable Patroni systemd unit.
+    4. Start Patroni systemd unit.
+3. Let Patroni take over the "startup procedure" of Postgres:
+    1. Restart all standby nodes.
+    2. Schedule a restart of the primary node during a maintenance window.
+4. Clean up slots from original cluster, if any:
+    1. Once Postgres is able to consume all WALs between the original slots and the new slots that have been created by Patroni, then remove the [permanent slots](https://patroni.readthedocs.io/en/latest/dynamic_configuration.html#:~:text=a%20standby%20leader-,slots,-%3A%20define%20permanent%20replication) configuration. This will allow Patroni to drop the original slots.
+
+**Note:** the above procedure was designed for minimal downtime. In any case we **strongly** recommend that you conduct these steps during a scheduled maintenance window. Also, we **strongly** recommend that you test at least a switchover/switchback operation after all the above steps have been performed.
+
+### More detailed example
+
+In this section we will cover an example with more detailed steps to migrate an existing Postgres cluster to a Patroni managed cluster.
+
+For sake of simplicity we will put some notes, when relevant, considering 3 possibilities for an existing cluster, just so we do not need to repeat ourselves regarding similar steps:
+
+1. Managed by [EFM](/efm/latest/).
+2. Managed by [repmgr](/repmgr/latest).
+3. With no HA tool.
+
+For the example, the following assumptions are made:
+
+1. You already have an working `etcd` cluster &mdash; refer to [Quick start on RHEL 8 - 1. Etcd](rhel8_quick_start#1-etcd)/[Quick start on Debian 11 - 1. Etcd](debian11_quick_start#1-etcd) for instructions on how to set up an `etcd` cluster. The `etcd` cluster here is composed of the following nodes:
```

Comment:
I would suggest `refer to [Installing and configuring etcd](installing_etcd) to install and set up etcd.`